server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -443,6 +443,26 @@ public String resolveDateMathExpression(String dateExpression) {
         return dateMathExpressionResolver.resolveExpression(dateExpression, new Context(null, null, isSystemIndexAccessAllowed()));
     }
 
+    /**
+     * @param time instant to consider when parsing the expression
+     * @return If the specified string is data math expression then this method returns the resolved expression.
+     */
+    public String resolveDateMathExpressionAtInstant(String dateExpression, long time) {
+        return dateMathExpressionResolver.resolveExpression(dateExpression, new Context(null, null, time, isSystemIndexAccessAllowed()));
+    }
+
+    /**
+     * @return If the specified string is data math expression then this method returns the resolved expression and the instant at which
+     * it was resolved
+     */
+    public ResolvedContext resolveDateMathExpressionAndInstant(String dateExpression) {
```

Comment:
This method and `ResolvedContext` class seems like not really needed?
The `resolveDateMathExpressionAndInstant(...)` method is only in `TransportCreateAction` and
there the `resolveDateMathExpressionAtInstant(...)` could be used? (using `System.currentTimeMillis()`,
and return value of `resolveDateMathExpressionAtInstant (...)` method)

Reply:
The rationale behind using the `ResolvedContext` instance is to create the index and aliases set in the index settings with the same timestamps, instead of using `System.currentTimeMillis()` for each

Reply:
That would work, yes.